src/ResourceManager/Sql/Commands.Sql/ManagedDatabase/Cmdlet/RestoreAzureSqlManagedDatabase.cs

```diff
@@ -0,0 +1,172 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Sql.ManagedDatabase.Model;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using System;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+
+namespace Microsoft.Azure.Commands.Sql.ManagedDatabase.Cmdlet
+{
+    [Cmdlet(VerbsData.Restore, "AzureRmSqlManagedDatabase", SupportsShouldProcess = true)]
+    public class RestoreAzureRmSqlManagedDatabase
+        : AzureSqlManagedDatabaseCmdletBase<AzureSqlManagedDatabaseModel>
+    {
+        protected const string PointInTimeRestoreFromNameAndResourceGroupParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromInputParameters";
+
+        protected const string PointInTimeRestoreFropmInputObjectParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromAzureSqlManagedDatabaseModelInstanceDefinition";
+
+        protected const string PointInTimeRestoreFromResourceIdParameterSet =
+            "PointInTimeRestoreManagedDatabaseFromAzureResourceId";
+
+        /// <summary>
+        /// Gets or sets flag indicating a restore from a point-in-time backup.
+        /// </summary>
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFromNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFropmInputObjectParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        [Parameter(
+            ParameterSetName = PointInTimeRestoreFromResourceIdParameterSet,
+            Mandatory = true,
+            HelpMessage = "Restore from a point-in-time backup.")]
+        public SwitchParameter FromPointInTimeBackup { get; set; }
+
+        /// <summary> 
+        /// Gets or sets the managed database name to restore
+        /// </summary>
+        [Parameter(ParameterSetName = PointInTimeRestoreFromNameAndResourceGroupParameterSet,
+            Mandatory = true,
+            Position = 0,
+            HelpMessage = "The managed database name to restore.")]
+        [Alias("Name")]
+        public string ManagedDatabaseName { get; set; }
```

Comment:
Rename this to Name, and switch order of the parameters to ResourceGroupName, ManagedInstanceName, then Name.